lib/Gallery.pm

```diff
@@ -159,5 +160,24 @@ sub resize_image {
 	return $new_name;
 }
 
+sub exifdate {
+	my ($image_path) = @_;
+
+	return 'unused' unless 'date' eq $config->{sort_images_by};
```

Comment:
Oh I see, because the Controller always sets the `date` field, even if this setting is off. Why not put the guard in Controller? (So don't set the `date` field if the setting is off, and thereby avoid the call to `exifdate`.) I suppose that code wouldn't make sense unless you knew that `exifdate` was going to try to avoid loading dependencies unless they are needed. Yuck either way.

You could resolve both issues by adding a `sortkey` field, and just setting it to whatever value the config specifies. Then `exifdate` would only be called when it was actually needed, and it could optimize itself the way it does without the caller having to be explicitly aware of the optimization. 

As always, as soon as a project get even a little bit complex, it needs unit tests :) (But that's not a request!)

I'm going to go ahead and accept the PR. One of us can fiddle with it later if we feel like it :)